test/functional/forall/atomic-view/tests/test-forall-AtomicMultiView.hpp

```diff
@@ -0,0 +1,128 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Header file containing basic functional tests for atomic operations with forall and views.
+///
+
+#ifndef __TEST_FORALL_ATOMIC_MULTIVIEW_HPP__
+#define __TEST_FORALL_ATOMIC_MULTIVIEW_HPP__
+
+#include <math.h>
+
+template <typename ExecPolicy,
+          typename AtomicPolicy,
+          typename WORKINGRES,
+          typename IdxType,
+          typename T>
+void ForallAtomicMultiViewTestImpl( IdxType N )
+{
+  // Functionally similar to ForallAtomicViewTestImpl
+
+  int dst_side = (int)(sqrt(N/2)); // dest[] dimension
+  int src_side = dst_side*2; // source[] dimension
+
+  RAJA::TypedRangeSegment<IdxType> seg(0, N);
+  RAJA::TypedRangeSegment<IdxType> seg_side(0, dst_side);
+
+  camp::resources::Resource work_res{WORKINGRES()};
+  camp::resources::Resource host_res{camp::resources::Host()};
+
+  T * actualsource = work_res.allocate<T>(N);
+  T * actualdest = work_res.allocate<T>(N/2);
+  T * check_array = host_res.allocate<T>(N/2);
+
+  // assumes each source[] will be 2x size of each dest[]
+  T ** source = new T * [src_side];
+  for ( int ii = 0; ii < src_side; ++ii )
+  {
+    source[ii] = actualsource+(ii*dst_side);
+  }
+
+  // assumes each dest[] will be a square matrix
+  T ** dest = new T * [dst_side];
+  for ( int ii = 0; ii < dst_side; ++ii )
+  {
+    dest[ii] = actualdest+(ii*dst_side);
+  }
+
+#if defined(RAJA_ENABLE_CUDA)
+  cudaErrchk(cudaDeviceSynchronize());
+#endif
+
+#if defined(RAJA_ENABLE_HIP)
+  hipErrchk(hipDeviceSynchronize());
+#endif
+
+  RAJA::forall<RAJA::seq_exec>(seg,
+                               [=](IdxType i) { actualsource[i] = (T)1; });
+
+  // use atomic add to reduce the array
+  // 1D defaut MultiView
+  RAJA::MultiView<T, RAJA::Layout<1>> vec_view(source, N);
+
+  // 1D MultiView with array-of-pointers index in 1st position
+  RAJA::MultiView<T, RAJA::Layout<1>, 1> sum_view(dest, N);
+  auto sum_atomic_view = RAJA::make_atomic_view<AtomicPolicy>(sum_view);
+
+
+  // Zero out dest using atomic MultiView
+  RAJA::forall<ExecPolicy>(seg_side, [=] RAJA_HOST_DEVICE(IdxType i) {
+    for ( int aopidx = 0; aopidx < dst_side; ++aopidx )
+    {
+      sum_atomic_view(i,aopidx) = (T)0;
+    }
+  });
+
+  // Assign values to dest using atomic MultiView
+  RAJA::forall<ExecPolicy>(seg_side, [=] RAJA_HOST_DEVICE(IdxType i) {
+    for ( int aopidx = 0; aopidx < src_side; ++aopidx )
+    {
+      sum_atomic_view(i, aopidx/2) += vec_view(aopidx,i);
+    }
+  });
+
+  work_res.memcpy( check_array, actualdest, sizeof(T) * N/2 );
+
+#if defined(RAJA_ENABLE_CUDA)
```

Comment:
We should have an abstraction for the sync operations that is a template on the execution back-end. I'll add an issue for this.

Reply:
It might also make sense to have a sync in the CAMP Resource, so we can just use something like `work_res.sync()`.

Reply:
We do have a templated synchronize method as @davidbeckingsale pointed out. I missed it.  I'm going to redefine the issue I added to use that in all tests, examples, etc.